service/service.go

```diff
@@ -12,87 +14,79 @@ import (
-	SearchAPIURL              string
-	SearchIndexProducer       kafka.Producer
-	ServiceAuthToken          string
-	Shutdown                  time.Duration
-	SignElasticsearchRequests bool
-	HasPrivateEndpoints       bool
+	Auditor                    audit.AuditorService
+	AuthAPIURL                 string
+	BindAddr                   string
+	DatasetAPIClient           api.DatasetAPIClient
+	DefaultMaxResults          int
+	Elasticsearch              api.Elasticsearcher
+	ElasticsearchURL           string
+	EnvMax                     int
+	HealthCheck                *healthcheck.HealthCheck
+	HealthCheckCriticalTimeout time.Duration
+	MaxRetries                 int
+	OutputQueue                searchoutputqueue.Output
+	SearchAPIURL               string
+	SearchIndexProducer        *kafka.Producer
+	ServiceAuthToken           string
+	Shutdown                   time.Duration
+	SignElasticsearchRequests  bool
+	HasPrivateEndpoints        bool
 }
 
 // Start handles consumption of events
-func (svc *Service) Start() {
+func (svc *Service) Start(ctx context.Context) {
 
 	signals := make(chan os.Signal, 1)
 	signal.Notify(signals, syscall.SIGINT, syscall.SIGTERM)
 
 	apiErrors := make(chan error, 1)
 
-	datasetAPINoAuth := dataset.NewAPIClient(svc.DatasetAPIURL, "", "")
-	datasetAPI := dataset.NewAPIClient(svc.DatasetAPIURL, svc.ServiceAuthToken, "")
-
-	healthTicker := healthcheck.NewTicker(
-		svc.HealthCheckInterval,
-		datasetAPINoAuth,
-		elasticsearch.NewHealthCheckClient(svc.ElasticsearchURL, svc.SignElasticsearchRequests),
-	)
+	svc.HealthCheck.Start(ctx)
 
 	api.CreateSearchAPI(
+		ctx,
 		svc.SearchAPIURL,
 		svc.BindAddr,
 		svc.AuthAPIURL,
 		apiErrors,
 		&svc.OutputQueue,
-		datasetAPI,
-		datasetAPINoAuth,
+		svc.DatasetAPIClient,
+		svc.ServiceAuthToken,
 		svc.Elasticsearch,
 		svc.DefaultMaxResults,
 		svc.HasPrivateEndpoints,
 		svc.Auditor,
+		svc.HealthCheck,
 	)
 
 	// blocks until a fatal error occurs
 	select {
 	case err := <-apiErrors:
-		log.ErrorC("api error received", err, nil)
+		log.Event(ctx, "api error received", log.ERROR, log.Error(err))
 	case <-signals:
-		log.Debug("os signal received", nil)
+		log.Event(ctx, "os signal received", log.INFO)
 	}
 
 	// Gracefully shutdown the application closing any open resources.
-	log.Info(fmt.Sprintf("shutdown with timeout: %s", svc.Shutdown), nil)
+	log.Event(ctx, fmt.Sprintf("shutdown with timeout: %s", svc.Shutdown), log.INFO)
 	ctx, cancel := context.WithTimeout(context.Background(), svc.Shutdown)
 
 	// stop any incoming requests before closing any outbound connections
 	api.Close(ctx)
 	if err := svc.SearchIndexProducer.Close(ctx); err != nil {
-		log.Error(errors.Wrap(err, "error while attempting to shutdown kafka producer"), nil)
+		log.Event(ctx, "error while attempting to shutdown kafka producer", log.ERROR, log.Error(err))
 	}
-	healthTicker.Close()
 
-	log.Info("shutdown complete", nil)
+	svc.HealthCheck.Stop()
+
+	log.Event(ctx, "shutdown complete", log.INFO)
 
```

Comment:
During graceful shutdown we should also Close the kafka producers (after the healthcheck)
if err := producer.Close(ctx) ...
if err := auditProducer.Close(ctx) ...